pkg/odo/cli/url/create.go

```diff
@@ -123,6 +134,24 @@ func (o *URLCreateOptions) Complete(_ string, cmd *cobra.Command, args []string)
 			} else {
 				o.urlType = envinfo.ROUTE
 			}
+			if len(o.protocol) == 0 {
+				// endpoint protocol default value is http
+				o.protocol = string(common.HTTP)
+			}
+			if len(o.exposure) == 0 {
+				// endpoint exposure default value is public
+				o.exposure = string(common.Public)
+			}
```

Comment:
set these defaults during flag initialization e.g
`			urlCreateCmd.Flags().StringVarP(&o.protocol, "protocol", "", string(common.HTTP), "protocol for this URL")
`

Reply:
updated